packages/react/collection/src/Collection.tsx

```diff
@@ -96,18 +96,19 @@ function createCollection<ItemElement extends HTMLElement, ItemData>() {
 
   function useCollection() {
     const context = React.useContext(Context);
-    return {
-      getItems() {
-        const orderedNodes = Array.from(
-          context.collectionRef.current!.querySelectorAll(`[${ITEM_DATA_ATTR}]`)
-        );
-        const items = Array.from(context.itemMap.values());
-        const orderedItems = items.sort(
-          (a, b) => orderedNodes.indexOf(a.ref.current!) - orderedNodes.indexOf(b.ref.current!)
-        );
-        return orderedItems;
-      },
-    };
+
+    const getItems = React.useCallback(() => {
+      const orderedNodes = Array.from(
+        context.collectionRef.current!.querySelectorAll(`[${ITEM_DATA_ATTR}]`)
```

Comment:
Something else that I was wondering was the non-null assertions. Right now the expectation is that you would know to only call this when the ref is available. If you try to get the collection inside a `useMemo` call or a `useLayoutEffect` it'll throw.

How do you feel about returning null so these cases can be handled at the consumer level?

Reply:
For this it might make sense to simply do this:

```tsx
context.collectionRef.current?.querySelectorAll(`[${ITEM_DATA_ATTR}]`) ?? []
```

no?

Reply:
Yup 👍